### PR TITLE
MLPAB-1101 - Spike - receive events and process

### PR DESCRIPTION
**Pull request by andrewpearce-digital**
# Purpose

Process events received from Sirius

Fixes MLPAB-1101

## Approach

- fix pipe name in IAM policy document
- use a rule to filter Sirius events sent to a reduced fees event bus
- queue the event (for now)
- update target event bus
- build a long lived environment called testevents
- enable reduced fees infra for all environments

## Learning

- https://aws.amazon.com/blogs/compute/reducing-custom-code-by-using-advanced-rules-in-amazon-eventbridge
- https://docs.aws.amazon.com/eventbridge/latest/userguide/eb-event-patterns-content-based-filtering.html
- https://docs.aws.amazon.com/eventbridge/latest/userguide/eb-events-structure.html
- https://docs.aws.amazon.com/eventbridge/latest/userguide/eb-event-bus-perms.html

### CODE REVIEW

File: terraform/environment/reduced_fees/README.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# Reduced Fees events
+
+This module creates event driven architecture to send and receive reduced fees events between MLPAB and Sirius.
+
+Outbound events are sent to the an Event Bridge event bus from DynamoDB Streams using EventBridge Pipes.
```
Comment (acsauk) [inline terraform/environment/reduced_fees/README.md:5]:
```suggestion
Outbound events are sent to an Event Bridge event bus from DynamoDB Streams using EventBridge Pipes.
```

File: terraform/environment/reduced_fees/main.tf [hunk 2]
```diff
@@ -195,3 +195,30 @@ resource "aws_cloudwatch_event_target" "cross_account_put" {
...
 }
+
+resource "aws_cloudwatch_event_rule" "receive_update" {
+  name           = "${data.aws_default_tags.current.tags.environment-name}-receive-update-more-evidence-required"
+  description    = "receive more evidence required update events from remote account"
+  event_bus_name = aws_cloudwatch_event_bus.reduced_fees.name
+
+  event_pattern = jsonencode({
+    source      = ["reduced-fees"],
+    detail-type = ["update"]
```
Comment (acsauk) [inline terraform/environment/reduced_fees/main.tf:206]: Do we need to include `detail` here as well or are the existing two keys sufficient for the pattern to match?

Comment (andrewpearce-digital) [reply terraform/environment/reduced_fees/main.tf:206]: This is sufficient for a match. It's what's required to see an event and identify what event bus it should be directed to. 

`detail` will contain some specific as yet undefined content that we should process, such as an item id, and a new status (or anything else the event producer needs to supply).

Comment (andrewpearce-digital) [reply terraform/environment/reduced_fees/main.tf:206]: There's information about filtering here https://docs.aws.amazon.com/eventbridge/latest/userguide/eb-event-patterns-content-based-filtering.html

**Review by acsauk — APPROVED** [review]
:shipit:

**Review by acsauk — APPROVED** [review]
:shipit:

**Pull request merged**